### ISSUE TITLE
Publish 1.2.3 at repo.maven.org

### ISSUE THREAD

**Issue by krichter722** [opencollab/flexdock#11]

**Comment by danilovesky** [comment]
@krichter722: Is it version 1.2.3 or 1.2.4 really? I noticed that there was a version bump in commit f12705f1934ea347b59f5f2befee156d29b0ea44. Also API changed a bit in version 1.2.4 (I had to tweak my code when moving from v 1.2.3).
PS: Thanks for supporting this repo!

**Comment by krichter722** [comment]
> @krichter722: Is it version 1.2.3 or 1.2.4 really? I noticed that there was a version bump in commit f12705f. Also API changed a bit in version 1.2.4 (I had to tweak my code when moving from v 1.2.3).

But I don't find `flexdock` when searching http://repo1.maven.org/maven2/ and http://repo2.maven.org/maven2/ or in the `org` subdirectory.

**Comment by danilovesky** [comment]
You are right, there is no maven package for flexdock. But I think the version of flexdock sources is actually 1.2.4, not 1.2.3 (as per log message in the above commit).

Note that I was using flexdock directly from https://forge.scilab.org/index.php/p/flexdock/ before I found this repository.

**Comment by krichter722** [comment]
> Note that I was using flexdock directly from https://forge.scilab.org/index.php/p/flexdock/ before I found this repository.

Do you mean you installed in after extracting the tarball or that you used this URL as maven repository. The latter is my intention because I'd like to minimize the efforts to build my project which depend on flexdock on CI services where I'd have to add a command chain to download, extract and install flexdock into the maven cache or create a maven aggregator project around my project which build flexdock before it builds my project.

It doesn't have to be the maven central repository, although it's state-of-the-art, but can be any maven repository for starters where releases are uploaded regularily. Thanks for you efforts for providing the builds out-of-the-box.

**Comment by danilovesky** [comment]
I was using a local just a local copy of prebuilt `flexdock-1.2.4.jar` file for my project. I totally agree it would be extremely useful to have flexdock in a repository. Note I am using gradle instead of maven for my projects, but the idea of a central repository for packages is the same :)

Note my initial comment was just about the version number for the produced flexdock jar files (not criticising anything really). I think it should be `flexdock-1.2.4.jar` (not `flexdock-1.2.3.jar`).

**Comment by krichter722** [comment]
I configured Sonatype snapshot and release deployment using [OSSRH](https://central.sonatype.org/pages/ossrh-guide.html). It takes about 1 hour to get everything up and running, approx. 3 hours to configure continuous deployment from Travis. If you have the intention to make flexdock available and easy to use for everyone this would be great contribution.

**Comment by krichter722** [comment]
This should work as expected after #13 is merged.

**Comment by sylvestre** [comment]
We can make a new dot release just for #13 change!

**Comment by krichter722** [comment]
> We can make a new dot release just for #13 change!

Go ahead :)

I pushlished 1.2.3 with my group ID `de.richtercloud` in accordance to the Apache 2.0 License as a temporary workaround.

**Comment by krichter722** [comment]
I just saw that 1.2.4 has already been released with f12705f1934ea347b59f5f2befee156d29b0ea44. So, the maven migration should be at least 1.2.5, maybe even 1.3 (not really sure what is best practice for changes to the build tool and project structure regarding version evolution) and #13 would be another revision. That should be reflected in the project before starting deployment to Maven central.

**Comment by sylvestre** [comment]
indeed. I just published 1.3
https://github.com/opencollab/flexdock/releases :)

**Comment by sylvestre** [comment]
I recreated the tag, I didn't updated everything, sorry :(

**Comment by sylvestre** [comment]
@krichter722 please let me know when you published it. I will update the readme

**Comment by krichter722** [comment]
I already published it as `de.richtercloud:flexdock-...`. `de.richtercloud` is the namespace I acquired at OSSRH, so it's a temporary workaround that I'd not like to do for another release. You need to acquire `org.flexdock`, please, by following the instructions linked above - I can't do that because I don't want to claim that namespace and administer it.

**Comment by crea-doo** [comment]
I also really appreciate the official publication of the artifacts on repo.maven.org.
I'm also willing to support this with a PR - so please let me know, if that's of interest for you.
